### DISCUSSION TITLE
Grafana how to show ip for RTCP packets

### DISCUSSION THREAD

**Discussion by boristh3blade** [sipcapture/homer#596] [Q&A]
Hello, I have latest homer7 docker installation, i collect data via heplify agent from asterisk servers
 it shows RTCP calls and i can filter it by node via {{node_id}}:
<img width="955" alt="node_id" src="https://github.com/sipcapture/homer/assets/49917738/4bccab51-7c0a-423d-95dc-9c9e151f49b9">

But cause i have a lot of connected phones to node I also want find what exactly call have trouble:
![node_id+cid](https://github.com/sipcapture/homer/assets/49917738/ed1b57b3-fcf4-47db-a24f-fad8fd6e822f)

To get this data i make such changes in metrics/prometeus.go file(original code commented):

```
        } else if pkt.ProtoType == 5 {
            if srcHit {
                //p.dissectRTCPStats(srcTarget, "src", pkt.NodeName, []byte(pkt.Payload))
                p.dissectRTCPStats(pkt.SrcIP, "src", pkt.NodeName, []byte(pkt.Payload))
            }
            if dstHit {
                //p.dissectRTCPStats(dstTarget, "dst", pkt.NodeName, []byte(pkt.Payload))
                p.dissectRTCPStats(pkt.DstIP, "dst", pkt.NodeName, []byte(pkt.Payload))

            }
            if !srcHit && !dstHit {
                //p.dissectRTCPStats("unknown", "", pkt.NodeName, []byte(pkt.Payload))
                p.dissectRTCPStats(pkt.CID, "direction-unknown", pkt.NodeName, []byte(pkt.Payload))
            }
```
Also to get srcIP, srcPort,dstIP,dstPort in Loki i update remotelog/loki.go :
 ```
          l.entry.labels["SrcIP"] = model.LabelValue(pkt.SrcIP)
            l.entry.labels["SrcPort"] =  model.LabelValue(strconv.FormatUint(uint64(pkt.SrcPort),10))
            l.entry.labels["DstIP"] = model.LabelValue(pkt.DstIP)
            l.entry.labels["DstPort"] =  model.LabelValue(strconv.FormatUint(uint64(pkt.DstPort),10))
            l.entry.labels["CorrelationID"] = model.LabelValue(pkt.CID)

```
So my question is, may be i miss something and exists simpler way for this ?
Cause now i need implement this changes in docker but im not very familiar with it.

**Comment by lmangani** [comment]
@boristh3blade please submit a PR with the changes so we can review their impact
